### DISCUSSION TITLE
redirect to different page after user login

### DISCUSSION THREAD

**Discussion by elenaaralla** [serenity-is/Serenity#7057] [Help] [answered]
Hi,

I would redirect user after successful login to a different page than dashboard depending on value of some user properties.
In which file I have to insert the control to decide which page load after login?

Thank you,
Elena

**Comment by furkanevran** [comment]
Hi @elenaaralla, you can create a controller that has a route of `~/` and redirect user accordingly.

For example:
```cs
public class IndexController : Controller
{ 
    [Route("~/")]
    public ActionResult Index([FromServices] ...)
    {
        // if user is not logged in redirect to login page using "return Redirect("~/Account/Login");" (your login page might be different)
        // get properties of user using ISqlConnections with IUserAccessor etc. and redirect to the according page
        // if user doesn't have a dashboard etc. return the fallback index file
    }
}
```

**Comment by Surenthiran** [comment]
Dashboardpage.cs is in Common/Dashboard page

this is my code for different Dashboard models 

`[Route("Dashboard/[action]")]
public class DashboardController : Controller
{

    private readonly ISqlConnections Connections;
    private IQmtHelper qmtHelper;
    UserDefinition userDef;
    public DashboardController(ISqlConnections connections, IUserAccessor userAccessor, IUserRetrieveService userRetrieveService, IQmtHelper qmtHelper)
    {
        this.Connections = connections ?? throw new ArgumentNullException(nameof(connections));
        userDef = userAccessor.User.GetUserDefinition<UserDefinition>(userRetrieveService);
        this.qmtHelper = qmtHelper ?? throw new ArgumentNullException(nameof(qmtHelper));

    }


    [PageAuthorize, HttpGet, Route("~/")]
    public ActionResult Index()
    {
        if (userDef.IsInRole("Employee"))
        {
            HrDashboardPageModel hrdashboardPageModel = new(Connections, userDef, qmtHelper);
            return View(MVC.Views.Common.EmployeeDashboard.EmployeeDashboardIndex, hrdashboardPageModel);
        }
        else if (userDef.IsInRole("Hr"))
        {
            HrDashboardPageModel hrdashboardPageModel = new(Connections, userDef, qmtHelper);
            return View(MVC.Views.Common.HrDashboard.HrDashboardIndex, hrdashboardPageModel);
        }else if (userDef.IsInRole("admin"))
        {
            DashboardPageModel admindashboardPageModel = new(Connections, userDef, qmtHelper);
            return View(MVC.Views.Common.Dashboard.DashboardIndex, admindashboardPageModel);
        }
        else
        {
            return View(MVC.Views.Common.Dashboard.DashboardIndex, null);
        }

        

    }`

**Reply by stixoffire** [reply]
This is EXACTLY what I was looking for and passing in the Model .was where I went wrong. Thank you for posting a real world example!

**Reply by stixoffire** [reply]
What namespace is ISqlConnections, IUserAccessor ?

**Reply by Surenthiran** [reply]
`using Serenity.Abstractions;
using Serenity.Data;`

**Comment by elenaaralla** [comment, marked as answer]
Thank you to all!

**Reply by Surenthiran** [reply]
if it answers your question, please mark it as answered
